### PR TITLE
Added dto and controller and all necessary objects

### PR DESCRIPTION
**Pull request by Petliuk**

### CODE REVIEW

File: src/main/java/com/example/mate/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/mate/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```

File: src/main/java/com/example/mate/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+public class BookController {
+
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/mate/controller/BookController.java]:
```suggestion
```

File: src/main/java/com/example/mate/controller/BookController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    private final BookService bookService;
+
+    @GetMapping
+    public List getAll() {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/mate/controller/BookController.java]: Parametrize list

File: src/main/java/com/example/mate/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        return bookService.getAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.getBookById(id);
+    }
+
+    @PostMapping
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/mate/controller/BookController.java:33]: ResponseStatus.CREATED

File: src/main/java/com/example/mate/controller/BookController.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
+        return bookService.createBook(bookDto);
+    }
+
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/mate/controller/BookController.java]:
```suggestion
```

File: src/main/java/com/example/mate/dto/BookDto.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package com.example.mate.dto;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/mate/dto/BookDto.java]: Don’t  use this annotation 
```suggestion
```

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  improve your solution ;)

File: src/main/java/com/example/mate/exception/EntityNotFoundException.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package com.example.mate.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.NOT_FOUND)
```
Comment (JJJazl) [inline src/main/java/com/example/mate/exception/EntityNotFoundException.java:6]: Interesting! I didn't know this works)
Don't forget to remove it when you add a global exception handler.

**Review by JJJazl — APPROVED** [review]

**Pull request merged**